js/bibox.js

```diff
@@ -1342,6 +1346,96 @@ module.exports = class bibox extends Exchange {
         };
     }
 
+    async transfer (code, amount, fromAccount, toAccount, params = {}) {
+        await this.loadMarkets ();
+        const currency = this.currency (code);
+        let method = undefined;
+        const request = {};
+        if (fromAccount === 'wallet' && toAccount === 'spot') {
+            method = 'privatePostAssetsTransferSpot';
+            request['type'] = 0;
+        } else if (fromAccount === 'spot' && toAccount === 'wallet') {
+            method = 'privatePostAssetsTransferSpot';
+            request['type'] = 1;
+        } else if (fromAccount === 'wallet' && toAccount === 'margin') {
+            method = 'privatePostCreditTransferAssetsBase2credit';
+        } else if (fromAccount === 'margin' && toAccount === 'wallet') {
+            method = 'privatePostCreditTransferAssetsCredit2base';
+        }
+        if (method === 'privatePostAssetsTransferSpot') {
+            request['symbol'] = currency['id'];
+            request['amount'] = this.currencyToPrecision (code, amount);
+        } else if (method === 'privatePostCreditTransferAssetsBase2credit' || method === 'privatePostCreditTransferAssetsCredit2base') {
```

Comment:
The code above could be condensed to 

```
if (fromAccount !== 'wallet' && toAccount !== 'wallet') {
            throw new BadRequest ('this exchange only allows transfers between wallet and spot, and between wallet and margin');
        }
        await this.loadMarkets ();
        const method = this.getSupportedMapping (fromAccount, {
            'wallet': this.getSupportedMapping (toAccount, {
                'spot': 'privatePostAssetsTransferSpot',
                'margin': 'privatePostCreditTransferAssetsBase2credit',
            }),
            'spot': 'privatePostAssetsTransferSpot',
            'margin': 'privatePostCreditTransferAssetsCredit2base',
        });
        const currency = this.currency (code);
        const request = {
            'amount': this.currencyToPrecision (code, amount),
        };
        if (method === 'privatePostAssetsTransferSpot') {
            request['type'] = fromAccount === 'wallet' ? 0 : 1;
            request['symbol'] = currency['id'];
        } else {
```

Reply:
Also, are you sure those are the only transfers that can be made? I see all these api endpoints for transfer

```
'transfer/mainAssets',
'transfer/transferIn',
'transfer/transferOut',
'transfer/transferInList',
'transfer/transferOutList',
'transfer/coinConfig',
'transfer/withdrawInfo',
'assets/transfer/spot',
'credit/transferAssets/base2credit',
'credit/transferAssets/credit2base',
'credit/transferAssets/lendAssets',
'credit/transferAssets/borrowAssets',
```

and with the web interface, It looks like I can transfer between any accounts

![](https://i.imgur.com/eRz8Bdy.png)

Reply:
Great comments! I'll address in seperate PR when adding `transfer` function